openfisca_core/parameters/parameter_scale.py

```diff
@@ -21,20 +21,27 @@ class ParameterScale(AtInstantLike):
     # 'unit' and 'reference' are only listed here for backward compatibility
     _allowed_keys = config.COMMON_KEYS.union({'brackets'})
 
-    def __init__(self, name, data, file_path):
+    def __init__(self, name, data = None, file_path = None, brackets = None):
         """
         :param name: name of the scale, eg "taxes.some_scale"
         :param data: Data loaded from a YAML file. In case of a reform, the data can also be created dynamically.
         :param file_path: File the parameter was loaded from.
         """
         self.name: str = name
         self.file_path: str = file_path
-        helpers._validate_parameter(self, data, data_type = dict, allowed_keys = self._allowed_keys)
         self.description: str = data.get('description')
         self.metadata: typing.Dict = {}
+
+        if brackets:
+            self.brackets = brackets
+
+        helpers._validate_parameter(self, data, data_type = dict, allowed_keys = self._allowed_keys)
         helpers._set_backward_compatibility_metadata(self, data)
         self.metadata.update(data.get('metadata', {}))
 
+        if brackets:
+            return
```

Comment:
@benjello How does this new behaviour relates to the one just below?

```python
        if not isinstance(data.get('brackets', []), list):
            raise ParameterParsingError(
                "Property 'brackets' of scale '{}' must be of type array."
                .format(self.name),
                self.file_path
                )

        brackets = []
        for i, bracket_data in enumerate(data.get('brackets', [])):
            bracket_name = helpers._compose_name(name, item_name = i)
            bracket = parameters.ParameterScaleBracket(name = bracket_name, data = bracket_data, file_path = file_path)
            brackets.append(bracket)
        self.brackets: typing.List[parameters.ParameterScaleBracket] = brackets
```

I;m just wondering whether the feature is already implemented but not used...